File: Data Structures & Algorithms/binary-tree-diameter/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    pair<int,int> diamf(TreeNode* root){
        if(root==NULL){
            return make_pair(0,0);
        } // diameter and height pair<int,int>
        pair<int,int> left=diamf(root->left);
        pair<int,int> right=diamf(root->right);
        int op1=left.first;
        int op2=right.first;
        int op3=left.second+right.second;
        pair<int,int> ans;
        ans.first=max(op1,max(op2,op3));
        ans.second=max(left.second,right.second)+1;
        return ans;
        
    }
    int diameterOfBinaryTree(TreeNode* root) {
        return diamf(root).first;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int s=0;
        int e=1;int ans=0;
        for(int i=e;i<prices.length;i++){
            if(prices[s]>prices[e]){
                s=e;
            }
            else{
                int p=prices[e]-prices[s];
                if(p>ans){
                    ans=p;
                }
            }
            e++;
        }
        return ans;
    }
}
